### PR TITLE
Make the post login event public

### PR DESCRIPTION
**Pull request by ChristophWurst**
So Mail and Suspicious Login can finally make use of it :)

### CODE REVIEW

**Review by gary-kim — APPROVED** [review]
Looks good to me

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -398,12 +399,12 @@ public function completeLogin(IUser $user, array $loginDetails, $regenerateSessi
...
 		}
 
-		$postLoginEvent = new OC\User\Events\PostLoginEvent(
+		$postLoginEvent = new PostLoginEvent(
 			$user,
 			$loginDetails['password'],
 			$isToken
 		);
-		$this->dispatcher->dispatch(OC\User\Events\PostLoginEvent::class, $postLoginEvent);
+		$this->dispatcher->dispatch(PostLoginEvent::class, $postLoginEvent);
```
Comment (juliushaertl) [inline lib/private/User/Session.php]:
```suggestion
		$this->dispatcher->dispatchTyped($postLoginEvent);
```

File: lib/public/User/Events/PostLoginEvent.php [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<?php
-declare(strict_types=1);
+<?php declare(strict_types=1);
```
Comment (juliushaertl) [inline lib/public/User/Events/PostLoginEvent.php]: More a styling thing: I think the declare statement has a dedicated line in most other places.

**Review by juliusknorr — APPROVED** [review]
Small nitpicks, otherwise :+1:

File: lib/public/User/Events/PostLoginEvent.php (hunk 2 above)
Comment (ChristophWurst) [reply lib/public/User/Events/PostLoginEvent.php]: Yeah, I do this sometimes. I can't find it right now but there was also a PSR that actually says the declare should be put *after* the license header :man_shrugging:

Comment (kesselb) [reply lib/public/User/Events/PostLoginEvent.php]: https://www.php-fig.org/psr/psr-12/

**Review by rullzer — APPROVED** [review]

File: lib/private/User/Session.php [hunk 3]
```diff
@@ -398,13 +399,11 @@ public function completeLogin(IUser $user, array $loginDetails, $regenerateSessi
 			$firstTimeLogin = $user->updateLastLoginTimestamp();
 		}
 
-		$postLoginEvent = new OC\User\Events\PostLoginEvent(
+		$this->dispatcher->dispatchTyped(new PostLoginEvent(
```
Comment (rullzer) [inline lib/private/User/Session.php]: Do we listen for it anywhere?
Should we make the private one just extend the public one for now? And deprecate it?

Comment (ChristophWurst) [reply lib/private/User/Session.php]: I'm not aware of anything. And I used phpunit's refactoring power to move the namespace. In the end it's private and thus we don't have to care :P

https://github.com/nextcloud/server/pull/16579 added it and https://github.com/nextcloud/suspicious_login/pull/146 is the only gh ticket referencing it.

Comment (ChristophWurst) [reply lib/private/User/Session.php]: https://github.com/search?q=org%3Anextcloud+PostLoginEvent&type=Code not used in our org

Comment (rullzer) [reply lib/private/User/Session.php]: ok then

**Comment by susnux** [comment]
@ChristophWurst what was the reason for this? Because there is already `UserLoggedInEvent` which is always emitted together with this event because `\OC\User::postLogin` hook is emitted so the server will emit `UserLoggedInEvent` because it listens on that hook.

**Comment by ChristophWurst** [comment]
I don't remember

**Comment by susnux** [comment]
Fair enough 6 years old change :sweat_smile:

**Pull request merged**